core/interop/src/lib.rs

```diff
@@ -390,6 +390,14 @@ impl<T: NativeObject> JsClass<T> {
     }
 }
 
+impl<T: NativeObject + Clone> JsClass<T> {
+    /// Clones the inner class instance.
+    #[must_use]
+    pub fn clone_inner(&self) -> T {
+        self.inner.borrow().data().clone()
+    }
+}
```

Comment:
Nit: this probably needs a `# Panics` comment specifying that it will panic if the inner data is already mutably borrowed.